API.md

```diff
@@ -0,0 +1,58 @@
+# SegM8Pi API
+
+## class SegM8
+
+Create an object of type `SegM8` to communicate with a chain of a particular [SegM8 7-segment
+indicator modules](https://my.amperka.com/modules/SegM8).
+
+### `SegM8(pin_CE: int, device_count: int = 1)`
+
+Construct a new SegM8 object that uses the default hardware SPI bus (SPI0).
+
+- `pin_CE`: the chip enable (also known as chip select or slave select) pin used to control the
+  shift-register latch. It can take the values **0** or **1**, which corresponds to the **CE0** or
+  **CE1** pins on the Raspberry Pi board.
+- `device_count`: the number of SegM8 modules connected in a daisy-chain. If omitted, defines
+  a single module.
+
+### `clear() -> None`
+
+Sets all module segments to the "dark" state. Clear the internal buffer.
+
+### `display(number: int, position: int, width: int, flags: int = SEGM8_ALIGN_RIGHT) -> None`
+
+### `display(number: float, position: int, width: int, precision: int = 1, flags: int = SEGM8_ALIGN_LEFT) -> None`
+
+### `display(string: str, position: int, width: int, flags: int = SEGM8_ALIGN_LEFT) -> None`
```

Comment:
:thinking: You cannot provide multiple methods with the same name in Python. What if:

```py
display_number(number, position, width, precision, align, pad, radix)
display_string(string, position, width)
```

?

Reply:
I was trying to simulate overloading using the `singledispathmethod` method. But it is only supported since python 3.8:(
So what you suggest is better and easier. Should I make a separate method for `float`?

Reply:
I think `display_number` is enough to serve both: ints and floats, but it has a little inconsistency between `radix` vs `precision` (only one makes sense in either case). So, it’s up to you, I have no strong preference.